src/corehost/common/pal.unix.cpp

```diff
@@ -170,6 +170,85 @@ bool pal::get_default_servicing_directory(string_t* recv)
     return true;
 }
 
+bool pal::get_local_shared_package_dir(pal::string_t* recv)
+{
+    recv->clear();
+    pal::string_t dir;
+    if (!pal::getenv("HOME", &dir))
+    {
+        struct passwd* pw = getpwuid(getuid());
+        if (pw && pw->pw_dir)
+        {
+            dir.assign(pw->pw_dir);
+        }
+    }
+    if (dir.empty())
+    {
+        return false;
+    }
+    append_path(&dir, _X(".dotnet"));
```

Comment:
Can you put this in a method called `get_local_dotnet_dir` taking an out parameter string to help @t-cakine 's work. Then append the arch shared package path.